docker/python3-scientific/create_zarr_optimus.py

```diff
@@ -224,6 +225,40 @@ def add_cell_metrics(data_group, input_path, cell_ids, verbose=False):
         logging.info("Not adding \"cell_metadata_numeric\" to zarr output: either the #genes or # cell ids is 0")
 
 
+def create_gene_id_name_map(gtf_file):
+    """ Creates a map from gene_id to gene_name by reading in the GTF file
+
+    Args:
+        annotation_file (string): annotation file 
+
+    Return:
+        gene_id_name_map (dict[string]->string): dictonary gene ids to gene names
+    """
+    gene_id_name_map = {}
+
+    try:
+        if gtf_file.endswith(".gz"):
+            fpin = gzip.open(gtf_file, 'rt')
+        else:
+            fpin = open(gtf_file, 'r')
+    except FileNotFoundError:
+        print("GTF File not found")
+
+        # loop through the lines and find the gene_id and gene_name pairs
+    for _line in fpin:
+        line = _line.strip()
+        gene_id_res = re.search(r'gene_id ([^;]*);', line)
+        gene_name_res = re.search(r'gene_name ([^;]*);', line)
+
+        if gene_id_res and gene_name_res:
+            gene_id = re.sub("\"", "", gene_id_res.group(1))
```

Comment:
Would be better to use single quotes rather than escaping: `'"'`. Also consider just using `replace` rather than `re.sub`; it's a little simpler.

Reply:
Agree! Fixed!